Friendly-plans/app/src/main/java/database/repository/ChildRepository.java

```diff
@@ -27,6 +28,25 @@ public long create(String firstName, String lastName) {
                 .list();
     }
 
+    public List<Child> getFilteredByFullName(String fullName) {
+        String[] splited = fullName.split("\\s+");
```

Comment:
I found that Greendao supports **like**. What not write only: 
``` 
return daoSession.getChildDao().queryBuilder()
                .whereOr(Properties.Name.like("%" + fullName + "%"),
                        Properties.Surname.like("%" + fullName + "%"))
                .list(); 
``` 
?

Reply:
Hi, do you have an idea how to write "WHERE (name LIKE ? AND surname LIKE ?) OR (name LIKE ? AND surname LIKE ?)"  this query using query builder?

Reply:
Why? What case are you trying to cover?

Reply:
"WHERE (name LIKE firstWord AND surname secondWord ?) OR (name secondWord ? AND surname LIKE firstWord)" - line 43 in ChildRepository.java

Reply:
Sorry, I didn't get your point. Sure, you need this case also.

code:
```
 QueryBuilder childQueryBuilder = daoSession.getChildDao().queryBuilder();
            childQueryBuilder.whereOr(childQueryBuilder.and(Properties.Name.like("%" + splited[0] + "%"),
                            Properties.Surname.like("%" + splited[1] + "%")),
                    childQueryBuilder.and(
                            Properties.Name.like("%" + splited[1] + "%"),
                            Properties.Surname.like("%" + splited[0] + "%"))
                    );
```

created sql ( you can see it in childQueryBuilder.build().sql): 
```
SELECT T."_id",T."NAME",T."SURNAME",T."FONT_SIZE",T."PICTURE_SIZE",T."DISPLAY_MODE" FROM "CHILD" T  
WHERE ((T."NAME" LIKE ? AND T."SURNAME" LIKE ?)
   OR (T."NAME" LIKE ? AND T."SURNAME" LIKE ?))
```